src/cpp/core/include/core/r_util/RTokenCursor.hpp

```diff
@@ -202,14 +209,14 @@ class RTokenCursor
       return currentToken().end();
    }
    
-   const RToken& nextToken() const
+   const RToken& nextToken(std::size_t offset = 1) const
```

Comment:
This will of course throw if the offset is out of range. Let's just make sure that we satisfy ourselves that everywhere we call this has no possibility of an out of bounds offset.